### PR TITLE
Update UG

### LINKED ISSUE #78: Update User Guide (up to v1.2)

### LINKED ISSUE #107: Update Clear Command
Update clear command to clear lesson list as well

### PR DESCRIPTION
**Pull request by fongyj**
- Organise and add commands to UG
- Update some javadocs
- Update HelpCommand and ClearCommand

Command descriptions in UG might not be complete yet and also can add screenshots as necessary.
Closes #78, Closes #107

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -126,118 +176,294 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
...
-**Clear** | `clear`
-**Delete** | `delete s/STUDENT_INDEX` or `delete t/TASK_INDEX` <br> e.g., `delete s/3` or `delete t/2`
-**Edit** | `edit INDEX [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS] [t/TAG]…​`<br> e.g.,`edit 2 n/James Lee e/jameslee@example.com`
-**Find** | `find KEYWORD [MORE_KEYWORDS]`<br> e.g., `find James Jake`
-**List** | `list s` or `list t`
-**Mark** | `mark t/TASK_INDEX` <br> e.g., `mark t/2`
-**Unmark** | `unmark t/TASK_INDEX` <br> e.g., `unmark t/2`
-**Mastery Check**| `mc MC_NUM s/STUDENT_INDEX r/RESULT` <br> e.g., `mc 1 s/1 r/PASS`
-**Help** | `help`
+Action | Format                                                           | Examples
```
Comment (fongyj) [inline docs/UserGuide.md:442]: It seems like in the original AB3 UG this table is sorted alphabetically, but I just put it according to the order above. Not sure if it has to be sorted alphabetically.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -126,118 +176,294 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
...
+
+Format: `addpart p/PARTICIPATION li/LESSON INDEX si/STUDENT_INDEX`
+
+* `PARTICIPATION` **must be a positive integer** from 0 to 500.
+* `LESSON_INDEX` refers to the index number shown in the displayed list of tasks.
+* `STUDENT_INDEX` refers to the index number shown in the student list for the specified lesson. (use `listlesson` to view the student list for lessons)
+* `LESSON_INDEX` and `STUDENT_INDEX` **must be a positive integer** 1, 2, 3, …​
+
+Example:
+* `addpart p/100 li/1 si/2` sets participation for the 2nd student from the 1st lesson to be 100.
```
Comment (laxus2308) [inline docs/UserGuide.md:410]: i think we should specify that the `Lesson` here must be a `Studio`, if for example `Mastery Check` is provided as LESSON_INDEX, the command will not work. 
```suggestion
### Adding participation for a student in a lesson : `addpart`

Adds participation for a student in a lesson.

Format: `addpart p/PARTICIPATION li/LESSON_INDEX si/STUDENT_INDEX`

* `PARTICIPATION` **must be a positive integer** from 0 to 500.
* `LESSON_INDEX` refers to the index number shown in the displayed list of tasks.
* `STUDENT_INDEX` refers to the index number shown in the student list for the specified lesson. (use `listlesson` to view the student list for lessons)
* `LESSON_INDEX` and `STUDENT_INDEX` **must be a positive integer** 1, 2, 3, …​

Example:
* `addpart p/100 li/1 si/2` sets participation for the 2nd student from the 1st lesson to be 100.
```

File: src/main/java/jarvis/logic/commands/AddParticipationCommand.java [hunk 3]
```diff
@@ -20,16 +20,17 @@
...
 public class AddParticipationCommand extends Command {
 
     public static final String COMMAND_WORD = "addpart";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Adds participation for a student for a studio. The student and studio are identified by their "
-            + "respective index number used in the displayed student list and in the displayed lesson list.\n"
+            + ": Adds participation for a student in a lesson. The lesson is identified by its index number in the"
+            + " displayed lesson list. The student is identified using its displayed index in the student list of the "
+            + "specified lesson\n"
```
Comment (laxus2308) [inline src/main/java/jarvis/logic/commands/AddParticipationCommand.java:33]:
```suggestion
            + ": Adds participation for a student in a studio. The studio is identified by its index number in the"
            + " displayed lesson list. The student is identified using its displayed index in the student list of the "
            + "specified studio\n"
```

File: docs/UserGuide.md [hunk 4]
```diff
@@ -126,118 +176,294 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
...
+**Add studio** | `addstudio [l/LESSON_DESC] sd/START_DATE st/START_TIME [ed/END_DATE] et/END_TIME` | `addconsult l/Consultation on recursion sd/2022-10-14 st/12:00 et/14:00 si/3 si/4`
+**Add mastery check** | `addmc [l/LESSON_DESC] sd/START_DATE st/START_TIME [ed/END_DATE] et/END_TIME si/STUDENT_INDEX...` | `addmc l/Mastery check 1 sd/2022-10-12 st/12:00 et/14:00 si/1 si/2`
+**Delete lesson** | `deletelesson INDEX` | `deletelesson 2`
+**Mark lesson** | `marklesson li/LESSON_INDEX` | `marklesson li/2`
+**Unmark lesson** | `unmarklesson li/LESSON_INDEX` | `unmarklesson li/2`
+**Mark student** | `markstudent li/LESSON_INDEX si/STUDENT_INDEX` | `markstudent li/2 si/3`
+**Unmark student** | `unmarkstudent li/LESSON_INDEX si/STUDENT_INDEX` | `unmarkstudent li/2 si/3`
+**Add note** | `addnote n/NOTE li/LESSON INDEX [si/STUDENT_INDEX]` | `addnote n/Get back to jeff on streams li/1 si/2`
+**Delete note** | `deletenote ni/NOTE INDEX li/LESSON INDEX [si/STUDENT_INDEX]` | `deletenote ni/1 li/1 si/2`
+**Add participation** | `addpart p/PARTICIPATION li/LESSON INDEX si/STUDENT_INDEX` |  `addpart p/100 li/1 si/2`
```
Comment (laxus2308) [inline docs/UserGuide.md:467]: missing the _ for some 

```suggestion
**Unmark student** | `unmarkstudent li/LESSON_INDEX si/STUDENT_INDEX` | `unmarkstudent li/2 si/3`
**Add note** | `addnote n/NOTE li/LESSON_INDEX [si/STUDENT_INDEX]` | `addnote n/Get back to jeff on streams li/1 si/2`
**Delete note** | `deletenote ni/NOTE_INDEX li/LESSON_INDEX [si/STUDENT_INDEX]` | `deletenote ni/1 li/1 si/2`
**Add participation** | `addpart p/PARTICIPATION li/LESSON_INDEX si/STUDENT_INDEX` |  `addpart p/100 li/1 si/2`
```

File: docs/UserGuide.md [hunk 5]
```diff
@@ -126,118 +176,294 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
...
+* `LESSON_INDEX` and `STUDENT_INDEX` **must be a positive integer** 1, 2, 3, …​
+
+Example:
+* `unmarkstudent li/2 si/3` marks the 3rd student in the 2nd lesson as absent.
+
+### Adding a note to a lesson : `addnote`
+
+Adds a note for a lesson or for a specific student in a lesson.
+
+Format: `addnote n/NOTE li/LESSON INDEX [si/STUDENT_INDEX]`
```
Comment (laxus2308) [inline docs/UserGuide.md:371]:
```suggestion
Format: `addnote n/NOTE li/LESSON_INDEX [si/STUDENT_INDEX]`
```

File: docs/UserGuide.md [hunk 6]
```diff
@@ -126,118 +176,294 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
...
+
+Example:
+* `addnote n/Get back to jeff on streams li/1 si/2` adds a note for the 2nd student in the 1st lesson.
+* `addnote n/Get back to the class on streams li/1` adds a note for the 1st lesson.
+
+### Deleting a note from a lesson : `deletenote`
+
+Deletes a note for a lesson or for a specific student in a lesson.
+
+Format: `deletenote ni/NOTE INDEX li/LESSON INDEX [si/STUDENT_INDEX]`
```
Comment (laxus2308) [inline docs/UserGuide.md:386]:
```suggestion
Format: `deletenote ni/NOTE_INDEX li/LESSON_INDEX [si/STUDENT_INDEX]`
```

File: src/main/java/jarvis/logic/commands/UnmarkStudentCommand.java [hunk 7]
```diff
@@ -18,16 +18,17 @@
...
     public static final String COMMAND_WORD = "unmarkstudent";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Marks a student as absent for a lesson. The student and lesson are identified by their respective"
-            + " index number used in the displayed student list and in the displayed lesson list.\n"
+            + ": Marks a student as absent for a lesson. "
+            + "The lesson is identified by its index number in the displayed lesson list. The student is identified "
+            + "using its displayed index in the student list of the specified lesson\n"
             + "Parameters: "
             + PREFIX_LESSON_INDEX + "LESSON INDEX "
```
Comment (laxus2308) [inline src/main/java/jarvis/logic/commands/UnmarkStudentCommand.java:33]:
```suggestion
            + PREFIX_LESSON_INDEX + "LESSON_INDEX "
```

**Review by laxus2308 — CHANGES_REQUESTED** [review]
Thanks for helping to update the UG! I spot some minor issues to make the style more consistent, and I think we should include in the command format how the format of deadline, date and time will be in our commands.

File: docs/UserGuide.md [hunk 8]
```diff
@@ -24,34 +24,37 @@ JARVIS is a **desktop app for a CS1101S Teaching Assistant to manage students an
...
 
 **:information_source: Notes about the command format:**<br>
 
 * Words in `UPPER_CASE` are the parameters to be supplied by the user.<br>
-  e.g. in `add s/STUDENT_NAME`, `NAME` is a parameter which can be used as `add s/John Tan`.
+  e.g. in `addstudent s/STUDENT_NAME m/MATRIC_NUM`, `STUDENT_NAME` and `MATRIC_NUM` are parameters which can be used as `addstudent s/John Doe m/A0123459G`.
 
 * Items in square brackets are optional.<br>
-  e.g `t/TASK_NAME [d/DATE]` can be used as `t/Mark quest 2 d/23-09-2023` or as `t/Mark quest 2`.
+  e.g `t/TASK_DESC [d/DEADLINE]` can be used as `t/Mark quest 2 d/23-09-2023` or as `t/Mark quest 2`.
```
Comment (neosunhan) [inline docs/UserGuide.md:51]: Task deadline should be in yyyy-MM-dd format

File: docs/UserGuide.md [hunk 9]
```diff
@@ -69,51 +75,95 @@ Shows a message explaning how to access the help page.
...
+
+Exits the program.
+
+Format: `exit`
+
+## Student features
+This section contains features for managing students.
 
-### Adding a student/task: `add`
+### Adding a student: `addstudent`
```
Comment (neosunhan) [inline docs/UserGuide.md:126]: I think we can include a line that says that you can only add up to 10 students

File: docs/UserGuide.md [hunk 10]
```diff
@@ -69,51 +75,95 @@ Shows a message explaning how to access the help page.
...
 * At least one of the optional fields must be provided.
 * Existing values will be updated to the input values.
-* When editing tags, the existing tags of the person will be removed i.e adding of tags is not cumulative.
-* You can remove all the person’s tags by typing `t/` without
-    specifying any tags after it.
 
 Examples:
-*  `edit 1 p/91234567 e/johndoe@example.com` Edits the phone number and email address of the 1st person to be `91234567` and `johndoe@example.com` respectively.
-*  `edit 2 n/Betsy Crower t/` Edits the name of the 2nd person to be `Betsy Crower` and clears all existing tags.
+*  `editstudent 1 s/John Do` edits the name the 1st student to be `John Do`
```
Comment (neosunhan) [inline docs/UserGuide.md:162]:
```suggestion
*  `editstudent 1 s/John Do` edits the name of the 1st student to be `John Do`
```

**Review by neosunhan — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md [hunk 11]
```diff
@@ -69,51 +75,96 @@ Shows a message explaning how to access the help page.
...
+Format: `addstudent s/NAME m/MATRIC_NUM`
 
+* `m/MATRIC_NUM` must start with 'A', followed by 7 digits and a capital letter
+* A maximum of 10 students can exist in the student list at any point in time. The user will not be allowed to add more students after reaching the limit of 10 students.
 
 Examples:
-* `add s/John Tan`
-* `add t/Consultation d/12-09-2023`
+* `addstudent s/John Doe m/A0123459G`
+* `addstudent s/Alex Yeoh m/A012345A`
```
Comment (rycs2812) [inline docs/UserGuide.md:137]: Matric number here is missing a digit

File: docs/UserGuide.md [hunk 12]
```diff
@@ -69,51 +75,96 @@ Shows a message explaning how to access the help page.
...
+* `deletestudent 2` deletes the 2nd student in JARVIS.
 
-### Editing a person : `edit`
+### Editing a student : `editstudent`
 
-Edits an existing person in the address book.
+Edits an existing student in JARVIS.
 
-Format: `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]…​`
+Format: `edit INDEX [s/NAME] [m/MATRIC_NUM]`
```
Comment (rycs2812) [inline docs/UserGuide.md:155]: Change to editstudent

File: docs/UserGuide.md [hunk 13]
```diff
@@ -69,51 +75,96 @@ Shows a message explaning how to access the help page.
...
-*  `edit 2 n/Betsy Crower t/` Edits the name of the 2nd person to be `Betsy Crower` and clears all existing tags.
+*  `editstudent 1 s/John Do` edits the name of the 1st student to be `John Do`
 
-### Locating persons by name: `find`
+### Locating students by name: `findstudent`
 
-Finds persons whose names contain any of the given keywords.
+Finds students whose names contain any of the given keywords.
 
 Format: `find KEYWORD [MORE_KEYWORDS]`
```
Comment (rycs2812) [inline docs/UserGuide.md:169]: Change to findstudent

File: docs/UserGuide.md [hunk 14]
```diff
@@ -126,118 +177,300 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
 
 Examples:
 * `find John` returns `john` and `John Doe`
```
Comment (rycs2812) [inline docs/UserGuide.md:179]: Change to findstudent

File: docs/UserGuide.md [hunk 15]
```diff
@@ -126,118 +177,300 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
...
+Adds a studio lesson to JARVIS.
 
-### Deleting a student/task : `delete`
+Format: `addstudio [l/LESSON_DESC] sd/START_DATE st/START_TIME [ed/END_DATE] et/END_TIME`
 
-Deletes the specified student or task from their respective lists.
+* `l/LESSON_DESC` is optional
+* `sd/START_DATE` and `ed/END_DATE` should be in the format `yyyy-MM-dd`.
+* `st/START_TIME` and `et/END_TIME` should be in the format `hh:mm`.
+* `ed/END_DATE` is optional, if unspecified `END_DATE` will be assumed to be the same as START_DATE
```
Comment (rycs2812) [inline docs/UserGuide.md:278]: Might be better to use inline code for START_DATE to be consistent i.e. change to `START_DATE`

File: docs/UserGuide.md [hunk 16]
```diff
@@ -126,118 +177,300 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
...
-* `STUDENT_INDEX` and `TASK_INDEX` refer to the index number shown in the respective displayed list.
-* `STUDENT_INDEX` and `TASK_INDEX` **must be a positive integer** 1, 2, 3, …​
+Adds a consult lesson to JARVIS.
+
+Format: `addconsult [l/LESSON_DESC] sd/START_DATE st/START_TIME [ed/END_DATE] et/END_TIME si/STUDENT_INDEX...`
+  
+* `l/LESSON_DESC` is optional
+* `sd/START_DATE` and `ed/END_DATE` should be in the format `yyyy-MM-dd`.
+* `st/START_TIME` and `et/END_TIME` should be in the format `hh:mm`.
+* `ed/END_DATE` is optional, if unspecified `END_DATE` will be assumed to be the same as START_DATE
```
Comment (rycs2812) [inline docs/UserGuide.md:292]: Same as above, change to `START_DATE`

File: docs/UserGuide.md [hunk 17]
```diff
@@ -126,118 +177,300 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
...
+* `deletenote ni/1 li/1 si/2` deletes a note for the 2nd student from the 1st lesson.
+* `deletenote ni/1 li/1` deletes a note from the 1st lesson.
+
+### Adding participation for a student in a lesson : `addpart`
+
+Adds participation for a student in a lesson.
+
+Format: `addpart p/PARTICIPATION li/LESSON_INDEX si/STUDENT_INDEX`
+
+* `PARTICIPATION` **must be a positive integer** from 0 to 500.
```
Comment (rycs2812) [inline docs/UserGuide.md:412]: Change "positive integer" to "non-negative integer"

File: docs/UserGuide.md [hunk 18]
```diff
@@ -126,118 +177,300 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
...
+**List students** | `liststudent`                                                                                     |
+**List tasks** | `listtask`                                                                                        |
+**List lessons** | `listlesson`                                                                                      |
+**List all** | `listall`                                                                                         |
+**Clear** | `clear`                                                                                           |
+**Add student** | `addstudent s/NAME m/MATRIC_NUM`                                                                  | `addstudent s/John Doe m/A0123459G`
+**Delete student** | `deletestudent INDEX`                                                                             | `deletestudent 2`
+**Edit student** | `edit INDEX [s/NAME] [m/MATRIC_NUM]`                                                              | `editstudent 1 s/John Do`
+**Find student** | `find KEYWORD [MORE_KEYWORDS]`                                                                    | `find John`
```
Comment (rycs2812) [inline docs/UserGuide.md:460]: Change to editstudent and findstudent

**Review by rycs2812 — CHANGES_REQUESTED** [review]
A few typos but everything is good apart from that

**Review by rycs2812 — APPROVED** [review]
LGTM

**Pull request merged**